ci/entrypoint.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+set -euo pipefail
+
+IBC_SRC=${IBC_SRC:-/src/ibc-rs}
+BASECOIN_SRC=${BASECOIN_SRC:-/src/basecoin-rs}
+BUILD_ROOT="${HOME}/build"
+IBC_BUILD="${BUILD_ROOT}/ibc-rs"
+BASECOIN_BUILD="${BUILD_ROOT}/basecoin-rs"
+BASECOIN_BIN="${BASECOIN_BUILD}/debug/tendermint-basecoin"
+HERMES_BIN="${IBC_BUILD}/release/hermes"
+IBC_REPO=https://github.com/informalsystems/ibc-rs.git
+IBC_COMMITISH=${IBC_COMMITISH:-master}
+CHAIN_DATA="${HOME}/data"
+HERMES_CONFIG="${HOME}/.hermes/config.toml"
+LOG_DIR=${LOG_DIR:-/var/log/basecoin-rs}
+TESTS_DIR=${TESTS_DIR:-${HOME}/tests}
+DEFAULT_TEST=${DEFAULT_TEST:-${TESTS_DIR}/create-connection.sh}
+
+if [ ! -f "${BASECOIN_SRC}/Cargo.toml" ]; then
+  echo "basecoin-rs sources must be mounted into ${BASECOIN_SRC} for this script to work properly."
+  exit 1
+fi
+
+if [ ! -f "${IBC_SRC}/Cargo.toml" ]; then
+  echo "No ibc-rs sources detected. Cloning repo at ${IBC_COMMITISH}..."
+  git clone "${IBC_REPO}" "${IBC_SRC}"
+  echo "Checking out ${IBC_COMMITISH}..."
+  cd "${IBC_SRC}"
+  git checkout "${IBC_COMMITISH}"
+  git status
+  echo ""
+fi
+
+cd "${IBC_SRC}"
+echo "Building Hermes..."
+cargo build --release --bin hermes --target-dir "${IBC_BUILD}/"
+
+cd "${BASECOIN_SRC}"
+echo ""
+echo "Building basecoin-rs..."
+cargo build --target-dir "${BASECOIN_BUILD}"
+
+echo ""
+echo "Setting up chain ibc-0..."
+mkdir -p "${CHAIN_DATA}"
+"${HOME}/one-chain" gaiad ibc-0 "${CHAIN_DATA}" 26657 26656 6060 9090 100000000000
+
+echo ""
+echo "Configuring Hermes..."
+"${HERMES_BIN}" -c "${HERMES_CONFIG}" \
+    keys add ibc-0 \
+    -f "${CHAIN_DATA}/ibc-0/user_seed.json"
+
+echo "Generating user key..."
+gaiad keys add user --keyring-backend="test" --output json > "${HOME}/user_seed.json"
+echo "Adding user key to basecoin-0 chain..."
+"${HERMES_BIN}" -c "${HERMES_CONFIG}" keys add basecoin-0 -f "${HOME}/user_seed.json"
```

Comment:
This command specifically fails for Gaiad v6.0.0, and where I suspect the ibc-rs serialization format may need to be updated. cc @adizere @hu55a1n1